zcash_client_sqlite/src/chain.rs

```diff
@@ -340,17 +339,12 @@ mod tests {
             &dfvk,
             AddressType::DefaultExternal,
             Amount::from_u64(7).unwrap(),
+            1,
         );
         insert_into_cache(&db_cache, &cb2);
 
-        // Data+cache chain should be valid
-        validate_chain(&db_cache, db_data.get_max_height_hash().unwrap(), None).unwrap();
-
         // Scan the cache again
-        scan_cached_blocks(&tests::network(), &db_cache, &mut db_data, None).unwrap();
-
-        // Data-only chain should be valid
-        validate_chain(&db_cache, db_data.get_max_height_hash().unwrap(), None).unwrap();
```

Comment:
Since `scan_cached_blocks` is performing this validation internally, I think these tests are still verifying that the conditions hold. It might be the case that they're now duplicative, but once we have out-of-order scanning I will want to check a bunch of chain validation conditions around that mechanic and this will be a good place to add such checks.